lib/util/source-code-util.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @fileoverview Tools for obtaining SourceCode objects.
+ * @author Ian VanSchooten
+ * @copyright 2016 Ian VanSchooten. All rights reserved.
+ * See LICENSE in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assign = require("object-assign"),
+    CLIEngine = require("../cli-engine"),
+    eslint = require("../eslint"),
+    globUtil = require("./glob-util"),
+    defaultOptions = require("../../conf/cli-options");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Get the SourceCode object for a single file
+ * @param   {string}     filename The fully resolved filename to get SourceCode from.
+ * @param   {Object}     options  A CLIEngine options object.
+ * @returns {SourceCode}          The SourceCode object representing the file.
+ */
+function getSourceCodeOfFile(filename, options) {
```

Comment:
There is a lot going on inside this function, is their any other better way to achieve this?
Can we run this for all file together rather then individual?